src/meta/src/stream/scale.rs

```diff
@@ -382,1985 +117,741 @@ impl ScaleController {
-                        .await?;
-
-                    parallelism.into()
-                }
-            };
-
-            job_reschedule_post_updates
-                .parallelism_updates
-                .insert(TableId::from(*job_id), parallelism);
-
-            let current_resource_group = match resource_group_update {
-                JobResourceGroupTarget::Update(Some(specific_resource_group)) => {
-                    job_reschedule_post_updates.resource_group_updates.insert(
-                        *job_id as ObjectId,
-                        Some(specific_resource_group.to_owned()),
-                    );
-
-                    specific_resource_group.to_owned()
-                }
-                JobResourceGroupTarget::Update(None) => {
-                    let database_resource_group = self
-                        .metadata_manager
-                        .catalog_controller
-                        .get_existing_job_database_resource_group(*job_id as _)
-                        .await?;
-
-                    job_reschedule_post_updates
-                        .resource_group_updates
-                        .insert(*job_id as ObjectId, None);
-                    database_resource_group
-                }
-                JobResourceGroupTarget::Keep => {
-                    self.metadata_manager
-                        .catalog_controller
-                        .get_existing_job_resource_group(*job_id as _)
-                        .await?
-                }
-            };
+        let reschedule = Reschedule {
+            added_actors,
+            removed_actors,
+            vnode_bitmap_updates,
+            upstream_fragment_dispatcher_ids,
+            upstream_dispatcher_mapping,
+            downstream_fragment_ids,
+            newly_created_actors,
+            actor_splits,
+            cdc_table_snapshot_split_assignment: Default::default(),
+            cdc_table_id: None,
+        };
 
-            let filtered_worker_ids =
-                filter_workers_by_resource_group(&workers, current_resource_group.as_str());
+        Ok(reschedule)
+    }
 
-            if filtered_worker_ids.is_empty() {
-                bail!("Cannot resize streaming_job {job_id} to empty worker set")
-            }
+    pub async fn reschedule_inplace(
+        &self,
+        policy: HashMap<ObjectId, ReschedulePolicy>,
+        workers: HashMap<WorkerId, PbWorkerNode>,
+    ) -> MetaResult<HashMap<DatabaseId, Command>> {
+        // jobs: HashMap<ObjectId, TargetResourcePolicy>,
+        // workers: BTreeMap<WorkerId, Worker>,
+        // ) -> MetaResult<HashMap<DatabaseId, HashMap<TableId, HashMap<FragmentId, crate::controller::fragment::InflightFragmentInfo >>>>
+
+        let inner = self.metadata_manager.catalog_controller.inner.read().await;
+        let txn = inner.db.begin().await?;
+
+        for (table_id, target) in &policy {
+            let streaming_job = StreamingJob::find_by_id(*table_id)
```

Comment:
Can we get all jobs and `max_parallelism` column only in one query for better performance?